### ISSUE TITLE
IMAGE_RESIZE_MODE = "pad64",   how to resize my images in rectangle and how it affects class_loss?

### ISSUE THREAD

**Issue by nemonameless** [matterport/Mask_RCNN#591]
hello, I  set IMAGE_RESIZE_MODE = "pad64" to train my images data, I just want my image to be rectangle and  not square,  such as 1280*960, but I find the mrcnn_class_loss descends quickly to under 0.1 ,  but my learning rate is 0.001, which is small enough.  Even if I set IMAGE_RESIZE_MODE = "pad64", and let IMAGE_MIN_DIM = 960, IMAGE_MAX_DIM = 960,  mrcnn_class_loss  also  descends quickly to under 0.1 ,  is it overfiting? 

how does the input size affect the class_loss ? and it seems that input size does not affect the box_loss, why?

And if I changed  to  IMAGE_RESIZE_MODE = "square" ,to be 1280*1280, it performs well . but I still want my input image to be rectangle, such as 1280*960 , then what can I do ?

**Comment by nemonameless** [comment]
@waleedka Hello, my images are not the same size, I just want my image to be rectangle and not square, such as 1280x960, so I use pad64 for training. 

And I find the images are resized to 1280x960 acctually, but when train my images  , the mrcnn_class_loss descends quickly to 0.1 and even smaller, it is abnormal ,but not seems to overfitting  ,the effect is very poor, especially for classification.   I want not use 'crop' to be rectangle. if I inference, I also want my images to be the same size as 1280x960. 

 It seems that the 'pad64'  is only for inference ???  then how can I resize my images in rectangle for train?thanks

**Comment by waleedka** [comment]
If your images are not the same size then pad64 can't be used for training. This mode pads images to the nearest multiple of 64 pixels but doesn't ensure that all images are resized to the same size. You need images to be of the same size so they can be stacked in a batch. For inferencing, though, you can use this mode as long as you use a batch size of 1.

You can do one of the following:
- Update the resize_image() function to return the rectangular shape you want. (recommended)
- Use IMAGE_RESIZE_MODE = "none" and do the resizing before passing the images to the model. For example, in your Dataset class.

**Comment by nemonameless** [comment]
@waleedka ok, thank you very much!

**Comment by georgezakinih** [comment]
Thanks for the great tool!
Does the image size for training has to be the same size of inference?
